tests/integration/test_task_events_mgr.py

```diff
@@ -99,3 +101,39 @@ async def test__insert_task_job(flow, one_conf, scheduler, start, validate):
             i.submit_num for i
             in schd.data_store_mgr.added['jobs'].values()
         ] == [1, 2]
+
+
+async def test__always_insert_task_job(
+    flow, one_conf, scheduler, start, validate
+):
+    """Insert Task Job _Always_ inserts a task into the data store.
+
+    Bug https://github.com/cylc/cylc-flow/issues/6172 was caused
+    by passing task state to data_store_mgr.insert_job: Where
+    a submission retry was in progress the task state would be
+    "waiting" which caused the data_store_mgr.insert_job
+    to return without adding the task to the data store.
+
+    n.b. future debuggers may with to convert the for loop to
+    pytest parameterization for ease of debugging, but the test is
+    very much faster with the parameterization inside.
+    """
+    conf = {
+        'scheduler': {'allow implicit tasks': True},
+        'scheduling': {'graph': {'R1': 'skarloey'}},
+    }
+    id_ = flow(conf)
```

Comment:
Can use the `one_conf` fixture already in the test sig:

```suggestion
    id_ = flow(one_conf)
```